sklearn/preprocessing/tests/test_encoders.py

```diff
@@ -637,3 +637,72 @@ def test_categories(density, drop):
 @pytest.mark.parametrize('Encoder', [OneHotEncoder, OrdinalEncoder])
 def test_encoders_has_categorical_tags(Encoder):
     assert 'categorical' in Encoder()._get_tags()['X_types']
+
+
+@pytest.mark.parametrize('Encoder', [OneHotEncoder, OrdinalEncoder])
+@pytest.mark.parametrize('col, categories', [
+    ('int_col', [3, 1, 2]),
+    ('str_col', ['d', 'z', 'u']),
+    ('float_col', [1.0, 3.1, 2.3]),
+])
+def test_multiple_pandas_category_no_order(Encoder, col, categories):
+    # order pandas categories will warn for ordered categories
+    pd = pytest.importorskip('pandas')
+
+    msg = (r"'auto' categories is used, but the Categorical dtype provided "
+           r"for column, {}, is not consistent with the automatic "
+           r"lexicographic ordering, lexicon order: {}, dtype order: {}. "
+           r"Consider passing a custom list of categories to the categories "
+           r"parameter.")
+
+    df = pd.DataFrame({'int_col': [1, 2, 3, 1, 1],
+                       'str_col': ['z', 'd', 'z', 'd', 'u'],
+                       'float_col':  [1, 2.3, 3.1, 1, 1]},
+                      columns=['int_col', 'str_col', 'float_col'])
+
+    cat_dtype = pd.api.types.CategoricalDtype(categories=categories,
+                                              ordered=True)
+    categories = cat_dtype.categories
+    custom_msg = msg.format(col,
+                            np.unique(df[col]),
+                            list(categories))
+    with pytest.warns(UserWarning) as record:
+        df_copy = df.assign(**{col: df[col].astype(cat_dtype)})
+        Encoder().fit(df_copy)
+    assert str(record[0].message) == custom_msg
+
+
+@pytest.mark.parametrize('Encoder', [OneHotEncoder, OrdinalEncoder])
+def test_pandas_category_same_lexicon_order(Encoder, ordered):
+    # does not warn when the order in the categories are the same as the
+    # lexicon ordering
```

Comment:
the order in the categories as defined by pandas? or by the user? or inferred by the encoder?

where does the lexicon order come from?